presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -208,1717 +207,1731 @@ public StatementAnalyzer(
         this.session = requireNonNull(session, "session is null");
     }
 
-    @Override
-    protected Scope visitUse(Use node, Scope scope)
+    public Scope analyze(Node node, Scope scope)
     {
-        throw new SemanticException(NOT_SUPPORTED, node, "USE statement is not supported");
+        return new Visitor().process(node, scope);
     }
 
-    @Override
-    protected Scope visitInsert(Insert insert, Scope scope)
+    private void analyzeWhere(Node node, Scope scope, Expression predicate)
     {
-        QualifiedObjectName targetTable = createQualifiedObjectName(session, insert, insert.getTarget());
-        if (metadata.getView(session, targetTable).isPresent()) {
-            throw new SemanticException(NOT_SUPPORTED, insert, "Inserting into views is not supported");
+        Visitor visitor = new Visitor();
+        visitor.analyzeWhere(node, scope, predicate);
+    }
+
+    private class Visitor
```

Comment:
I don't think it should be static (see `ExpressionAnalyzer.Visitor`). It uses fields from outer class